pkg/pollers/nodePoller.go

```diff
@@ -171,7 +174,9 @@ func (np *nodePoller) poller() {
 
 		if true == doPoll {
 			doPoll = false
-			nodes, err := np.kubeClient.Core().Nodes().List(metav1.ListOptions{})
+
+			// LabelSelector
+			nodes, err := np.kubeClient.Core().Nodes().List(metav1.ListOptions{LabelSelector: np.nodeLabel})
```

Comment:
If nodeLabel is an empty string, will this default to watching all nodes? Have you tested this?

Reply:
We found the test for it. Looks good.